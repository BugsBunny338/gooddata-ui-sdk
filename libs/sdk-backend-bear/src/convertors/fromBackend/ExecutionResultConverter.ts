// (C) 2019-2020 GoodData Corporation

import { GdcExecution } from "@gooddata/api-model-bear";
import { IResultWarning } from "@gooddata/sdk-backend-spi";
import { isUri } from "@gooddata/api-client-bear";
import { uriRef } from "@gooddata/sdk-model";

export function convertWarning(warning: GdcExecution.Warning): IResultWarning {
    return {
        warningCode: warning.warningCode,
        message: warning.message,
        parameters: warning.parameters?.map((param) => (isUri(param) ? uriRef(param) : param)),
    };
}
