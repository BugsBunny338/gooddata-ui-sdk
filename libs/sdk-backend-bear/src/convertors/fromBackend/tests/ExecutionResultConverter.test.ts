// (C) 2020 GoodData Corporation

import { convertDimensions } from "../ExecutionResultConverter";
import { dimensions } from "./ExecutionResultConverter.fixtures";

describe("convertDimensions", () => {
    it("should add ref properties as uriRefs", () => {
        expect(convertDimensions(dimensions)).toMatchSnapshot();
    });
});
