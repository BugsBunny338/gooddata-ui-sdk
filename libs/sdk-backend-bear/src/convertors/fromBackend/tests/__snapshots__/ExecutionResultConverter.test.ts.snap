// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertDimensions should add ref properties as uriRefs 1`] = `
Array [
  Object {
    "headers": Array [
      Object {
        "measureGroupHeader": Object {
          "items": Array [
            Object {
              "measureHeaderItem": Object {
                "format": "#,##0.00",
                "identifier": "af2Ewj9Re2vK",
                "localIdentifier": "lostMetric",
                "name": "<button>Lost</button> ...",
                "ref": Object {
                  "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1283",
                },
                "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1283",
              },
            },
            Object {
              "measureHeaderItem": Object {
                "format": "#,##0.00",
                "identifier": "afSEwRwdbMeQ",
                "localIdentifier": "wonMetric",
                "name": "Won",
                "ref": Object {
                  "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1284",
                },
                "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1284",
              },
            },
            Object {
              "measureHeaderItem": Object {
                "format": "#,##0.00",
                "identifier": "alUEwmBtbwSh",
                "localIdentifier": "expectedMetric",
                "name": "Expected",
                "ref": Object {
                  "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1285",
                },
                "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/1285",
              },
            },
          ],
          "totalItems": undefined,
        },
      },
    ],
  },
  Object {
    "headers": Array [
      Object {
        "attributeHeader": Object {
          "formOf": Object {
            "identifier": "created",
            "name": "Year created",
            "ref": Object {
              "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/157",
            },
            "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/157",
          },
          "identifier": "created.aag81lMifn6q",
          "localIdentifier": "yearCreatedAttribute",
          "name": "Year (Created)",
          "ref": Object {
            "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/158",
          },
          "uri": "/gdc/md/d20eyb3wfs0xe5l0lfscdnrnyhq1t42q/obj/158",
        },
      },
    ],
  },
]
`;
